src/adapter.ts

```diff
@@ -72,10 +103,10 @@ class MongooseAdapter {
    * const adapter = await MongooseAdapter.newAdapter('MONGO_URI');
    * const adapter = await MongooseAdapter.newAdapter('MONGO_URI', { mongoose_options: 'here' });
    */
-  static async newAdapter (uri, options = {}, adapterOptions = {}) {
+  static async newAdapter(uri: string, options: ConnectOptions = {}, adapterOptions: MongooseAdapterOptions = {}) {
     const adapter = new MongooseAdapter(uri, options);
     await adapter._open();
```

Comment:
[As discussed here](https://github.com/node-casbin/mongoose-adapter/pull/26#discussion_r607150009): Maybe just remove this function call and add it as a one-liner `this.connection = await connect(this.uri, this.options)`